tooling/image-sync/go.mod

```diff
@@ -0,0 +1,88 @@
+module github.com/Azure/ARO-HCP/tooling/image-sync
+
+go 1.22.2
+
+require (
+	github.com/Azure/azure-sdk-for-go/sdk/azcore v1.12.0
+	github.com/Azure/azure-sdk-for-go/sdk/azidentity v1.7.0
+	github.com/Azure/azure-sdk-for-go/sdk/containers/azcontainerregistry v0.2.1
+	github.com/containers/image v3.0.2+incompatible
+	github.com/spf13/cobra v1.8.1
+	github.com/spf13/viper v1.19.0
+	go.uber.org/zap v1.27.0
```

Comment:
according to docs [here](https://github.com/uber-go/zap), it is stated that

> Note that zap only supports the two most recent minor versions of Go.

this is a vague statement, but do we anticipate anything breaking due to this?

Reply:
I Don't think it's an issue, cause we can pin the version.

WDYT?

Reply:
> I Don't think it's an issue, cause we can pin the version.
> 
> WDYT?

fine with me then